File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        uniqueSolution=[]
        incomingDegree=[False]*n
        for edge in edges:
            incomingDegree[edge[1]]=True
        for index in range(n):
            if not incomingDegree[index]:
                uniqueSolution.append(index)
        return uniqueSolution
        
        

File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        List<Integer> uniqueSolution=new ArrayList();
        boolean[] incomingDegree=new boolean[n];
        for(int i=0;i<edges.size();i++){
            incomingDegree[edges.get(i).get(1)]=true;
        }
        for(int i=0;i<n;i++){
            if(!incomingDegree[i]){
                uniqueSolution.add(i);
            }
        }
        return uniqueSolution;
    }
}